service/service_lifecycle_manager.go

```diff
@@ -0,0 +1,62 @@
+package service
+
+import (
+	"github.com/vmware/transport-go/model"
+	"net/http"
+)
+
+var svcLifecycleManagerInstance ServiceLifecycleManager
+
+type ServiceLifecycleManager interface {
+	GetServiceHooks(serviceChannelName string) ServiceLifecycleHookEnabled
+}
+
+type ServiceLifecycleHookEnabled interface {
+	OnServiceReady() chan struct{}            // service initialization logic should be implemented here
```

Comment:
why is this a `struct{}` a not just a `bool`? does this need to do anything other than just say `I am ready`

Reply:
i updated it to return `chan bool`.